docs/source/examples/example_dates.rst

```diff
@@ -0,0 +1,158 @@
+3. The minimum value in column ``date_from`` should be the first of January 2022.
+4. The maximum value in column ``date_to`` should be the 31st of January 2022.
+5. There is no gap between ``date_from`` and ``date_to``. In other words, every date
+   of January can be assigned to at least one row.
+6. There is no overlap between ``date_from`` and ``date_to``. In other words, every
+   date of January can be assigned to at most one row.
+
+
+Assuming that such a table exists in database, we can write a specification against it.
+
+.. code-block:: python
+
+    import pytest
+    import sqlalchemy as sa
+
+    from datajudge import WithinRequirement
+    from datajudge.pytest_integration import collect_data_tests
+
+    # We create a Requirement, within a table. This object will contain
+    # all the constraints we want to test on the specified table.
+    # To test another table or test the same table against another table,
+    # we would create another Requirement object.
+    prices_req = WithinRequirement.from_table(
+        db_name="example", schema_name=None, table_name="prices"
+    )
+
+    # Constraint 1:
+    # All values from column date_from should be in January 2022.
+    prices_req.add_date_between_constraint(
+        column="date_from",
+	lower_bound="'20220101'",
+	upper_bound="'20220101'",
+	# We don't tolerate any violations of the constraint:
+	min_fraction=1,
+    )
+
+    # Constraint 2:
+    # All values from column date_to should be in January 2022.
+    prices_req.add_date_between_constraint(
+        column="date_to",
+	lower_bound="'20220101'",
+	upper_bound="'20220101'",
+	# We don't tolerate any violations of the constraint:
+	min_fraction=1,
+    )
+
+    # Constraint 3:
+    # The minimum value in column date_from should be the first of January 2022.
+
+    # Ensure that the minimum is smaller or equal the reference value min_value.
+    prices_req.add_date_min_constraint(column="date_from", min_value="'20220101'")
+    # Ensure that the minimum is greater or equal the reference value min_value.
+    prices_req.add_date_min_constraint(
+        column="date_from",
+	min_value="'20220101'",
+	use_upper_bound_reference=True,
+    )
+
+    # Constraint 4:
+    # The maximum value in column date_to should be the 31st of January 2022.
+
+    # Ensure that the maximum is greater or equal the reference value max_value.
+    prices_req.add_date_max_constraint(column="date_to", max_value="'20220131'")
+    # Ensure that the maximum is smaller or equal the reference value max_value.
+    prices_req.add_date_max_constraint(
+        column="date_to",
+	max_value="'20220131'",
+	use_upper_bound_reference=True,
+    )
+
+    # Constraint 5:
+    # There is no gap between date_from and date_to. In other words, every date
+    # of January can be assigned to at least one row.
+    prices_req.add_date_no_gap_constraint(
+        start_column="date_from",
+	end_column="date_to",
+	key_columns=["price"],
```

Comment:
Maybe you want to explain here the role of this argument.

Reply:
Yeah good point! This was actually a mistake - fixed now.